### PR TITLE
Added cpuInfo to libXBMC_addon

### PR DESCRIPTION
**Pull request by AchimTuran**
I added the cpuInfo methods to libXBMC_addon. At the moment I only need CPUInfo_GetCPUFeatures() to get CPUFeatures bits (SSE, ...). I need this for the new adsp addons to activate or deactivate my optimized processing functions for SSE, AVX, ...

Should I also add "CPUInfo_GetTemperature" and "CPUInfo_GetCoreInfo"?
How do we get the CPU Temperature on Windows? I don't see any functionality in "CPUInfo.cpp".

### CODE REVIEW

**Comment by AchimTuran** [comment]
What can I do with CPUInfo.GetCPUBogoMips()?

File: xbmc/addons/AddonCallbacksAddon.cpp [hunk 1]
```diff
@@ -529,4 +546,121 @@ bool CAddonCallbacksAddon::RemoveDirectory(const void* addonData, const char *st
...
+  return g_cpuInfo.getCPUCount();
+}
+
+float CAddonCallbacksAddon::CPUInfo_GetCPUFrequency(const void* addonData)
+{
+  CAddonCallbacks* helper = (CAddonCallbacks*)addonData;
+  if (!helper)
+    return false;
+
+  return g_cpuInfo.GetCPUFeatures();
```
Comment (AlwinEsch) [inline xbmc/addons/AddonCallbacksAddon.cpp]: Here the wrong function becomes called.

File: xbmc/utils/CPUFeatures.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#define CPU_FEATURE_SSE      1 << 2
+#define CPU_FEATURE_SSE2     1 << 3
+#define CPU_FEATURE_SSE3     1 << 4
+#define CPU_FEATURE_SSSE3    1 << 5
+#define CPU_FEATURE_SSE4     1 << 6
+#define CPU_FEATURE_SSE42    1 << 7
+#define CPU_FEATURE_3DNOW    1 << 8
+#define CPU_FEATURE_3DNOWEXT 1 << 9
+#define CPU_FEATURE_ALTIVEC  1 << 10
+#define CPU_FEATURE_NEON     1 << 11
```
Comment (AlwinEsch) [inline xbmc/utils/CPUFeatures.h]: Think on empty line on end ;)

Comment (AchimTuran) [inline xbmc/utils/CPUFeatures.h]: Thanks, fixed.

File: xbmc/addons/AddonCallbacksAddon.cpp (hunk 1 above)
Comment (AchimTuran) [inline xbmc/addons/AddonCallbacksAddon.cpp]: Thanks, fixed.

**Comment by AchimTuran** [comment]
> If something like yatse or another remote wants to display a systeminfo, there would be a usecase, but that's only my personal uneducated guess.

I removed it, if we need it I will add it again.


If all is ok, I will squash all commits.

**Comment by AchimTuran** [comment]
Some question about function calling:
1. XBMC->CPUInfo_GetCPUFeatures()
2. XBMC->CPUInfo.GetCPUFeatures() (implementing a between class)

What do you prefer 1 or 2?

File: xbmc/utils/CPUInfo.h [hunk 3]
```diff
@@ -35,18 +36,19 @@ typedef HANDLE PDH_HCOUNTER;
...
+//#define CPU_FEATURE_SSE      1 << 2
+//#define CPU_FEATURE_SSE2     1 << 3
+//#define CPU_FEATURE_SSE3     1 << 4
+//#define CPU_FEATURE_SSSE3    1 << 5
+//#define CPU_FEATURE_SSE4     1 << 6
+//#define CPU_FEATURE_SSE42    1 << 7
+//#define CPU_FEATURE_3DNOW    1 << 8
+//#define CPU_FEATURE_3DNOWEXT 1 << 9
+//#define CPU_FEATURE_ALTIVEC  1 << 10
+//#define CPU_FEATURE_NEON     1 << 11
```
Comment (AlwinEsch) [inline xbmc/utils/CPUInfo.h]: Can be removed if on CPUFeatures.h

**Comment by wsnipex** [comment]
wouldn't it make sense to add those to https://github.com/xbmc/xbmc/blob/master/xbmc/utils/SystemInfo.cpp

**Comment by AchimTuran** [comment]
I think no, because we also need those define's at the addon side. So we only have one file to maintain.

**Comment by AchimTuran** [comment]
Please close this PR, because I will add cpuInfo to kodi-utilities or as separate static library.
If you want here is my [current](https://github.com/AchimTuran/asplib) work.

**Pull request closed without merging**